src/data-access/repositories/mongo/mongo-entity-converter.ts

```diff
@@ -115,29 +132,77 @@ interface MongoLayerMapping {
 }
 
 export class MongoEntityConverter {
-  public convertRundown(mongoRundown: MongoRundown, baselineTimelineObjects?: TimelineObject[]): Rundown {
+  public convertRundown(mongoRundown: MongoRundown, segments: Segment[], baselineTimelineObjects?: TimelineObject[], infinitePieces?: Piece[]): Rundown {
+    let alreadyActiveProperties: RundownAlreadyActiveProperties | undefined
+    if (mongoRundown.isActive) {
+      alreadyActiveProperties = {
+        activeCursor: this.convertMongoRundownCursorToRundownCursor(mongoRundown.activeCursor, segments),
+        nextCursor: this.convertMongoRundownCursorToRundownCursor(mongoRundown.nextCursor, segments),
+        infinitePieces: this.mapToInfinitePieceMap(infinitePieces ?? [])
+      }
+    }
```

Comment:
We should use immutable variables as much as possible in calculations to make the flow deterministic and hereby easier to understand. When I see a let binding, I expect (and fear) that the value is replaced during its lifetime.
In this case you can make it a ternary (like below) or if that becomes too messy, you can extract to a separate method.
```typescript
    const alreadyActiveProperties: RundownAlreadyActiveProperties | undefined = !mongoRundown.isActive ? undefined : {
      activeCursor: this.convertMongoRundownCursorToRundownCursor(mongoRundown.activeCursor, segments),
      nextCursor: this.convertMongoRundownCursorToRundownCursor(mongoRundown.nextCursor, segments),
      infinitePieces: this.mapToInfinitePieceMap(infinitePieces ?? [])
    }
```

Reply:
I don't share you fear, but it has been updated :)